concrete/controllers/dialog/page/add/compose.php

```diff
@@ -93,7 +95,12 @@ public function submit()
                     $dateTime = new DateTime();
                     $publishDateTime = $dateTime->translate('cvPublishDate');
                     $publishEndDateTime = $dateTime->translate('cvPublishEndDate');
-                    if ($this->request->request->get('keepOtherScheduling')) {
+                    $app = Application::getFacadeApplication();
+                    $appConfig = $app->make(Repository::class);
+                    $liveVersionStatusOnScheduledVersionApproval = (string)$appConfig->get('concrete.misc.live_version_status_on_scheduled_version_approval');
+                    $isUnapproved = $liveVersionStatusOnScheduledVersionApproval === 'unapproved';
+                    $isKeepOtherScheduling = (bool)$this->request->request->get('keepOtherScheduling');
+                    if (($isUnapproved && !$isKeepOtherScheduling) || (!$isUnapproved && $isKeepOtherScheduling)) {
```

Comment:
What about simply

```php
if ($isUnapproved !== !$isKeepOtherScheduling) {
```

Reply:
@mlocati you mean
`if ($isUnapproved === !$isKeepOtherScheduling) {` makes a lot of sense, i was too much into the xor :)

Reply:
> What about simply
>
> ```php
> if ($isUnapproved !== !$isKeepOtherScheduling) {
> ```

Well, I actually meant

```php
if ($isUnapproved !== $isKeepOtherScheduling) {
```

Reply:
:D Now we are in deep cosmetic. But i can also change to that, it's fast done?

Reply:
No need to update this PR! I was just blaming myself for my mistake 😉